### PR TITLE
chore : get repo in serverside

### PR DESCRIPTION
**Pull request by sifatul**
TODO: 
1. create api for user info using access token (similar to /api/platform/github/repos)
2. update api calls in github-area.tsx

### CODE REVIEW

**Comment by fazlerabby07** [comment]
@sifatul create api for github users and update github api in githun.tsx.
find this problem for medium
image

**Comment by sifatul** [comment]
> @sifatul create api for github users and update github api in githun.tsx. find this problem for medium image

resolved

File: package.json [hunk 1]
```diff
@@ -19,13 +19,15 @@
     "eslint-config-prettier": "^8.5.0",
     "graphql": "^16.5.0",
     "graphql-request": "^4.3.0",
+    "http": "^0.0.1-security",
```
[the remark this answers is not in the document]
Comment (fazlerabby07) [reply package.json:22]: okey. i wil remove it.

File: pages/api/platform/github/repos.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+const handler = nextConnect();
+export default handler;
+
+const getRepoList = async (req: NextApiRequest, res: NextApiResponse<any>) => {
+    const username = req.query.username || '';
+    if (!username) res.status(404).end();
+    const getRepoListApi = `GET /users/${username}/repos`;
+    const octokit = new Octokit({ auth: process.env.GITHUB_ACCESS_TOKEN });
+    const githubResponse = await octokit.request(getRepoListApi, {});
+    const repos = githubResponse?.data;
```
[the remark this answers is not in the document]
Comment (fazlerabby07) [reply pages/api/platform/github/repos.ts:14]: @sifatul bhai, those who have  more then 100 repository, git repository url give 30 repository list. If we have to get more we can use like this `https://api.github.com/users/fazlerabby07/repos?per_page=30&page=1`

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (fazlerabby07) [reply package.json]: @sifatul done

File: pages/api/platform/github/repos.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (sifatul) [reply pages/api/platform/github/repos.ts:14]: @fazlerabby07 please create another issue for working with paging.

**Pull request merged**